expertsystem/data.py

```diff
@@ -320,8 +320,23 @@ def __init__(self, particles: Optional[Iterable[Particle]] = None) -> None:
                 }
             )
 
-    def __contains__(self, particle_name: object) -> bool:
-        return particle_name in self.__particles
+    def __contains__(self, instance: object) -> bool:
+        if isinstance(instance, str):
+            return instance in self.__particles
+        if isinstance(instance, Particle):
+            return instance in self.__particles.values()
```

Comment:
This could be `return instance.name in self.__particles`

Reply:
No, because it checks whether the `Particle` instance is in the `ParticleCollection` 'set'. It's also tested here
https://github.com/ComPWA/expertsystem/blob/f2323fe1bd079ea41b37a2565bc431ba36d0be67/tests/unit/test_data.py#L269

Reply:
Well its the same thing (just faster), as you guarantee that the string keys are unique and match their particles.

This actually touches an issue. What if you define a "faulty" particle with different properties, but the same name as an existing particle in the collection. Should that return a `True` or a `False` for `__contains__`?

Reply:
That is true. I had a different check in mind, so maybe you're right. But technically, for a `set` only `instance in self.__particles.values()` should be the correct behaviour.

Reply:
With #295 implemented, is the existing implementation correct?

Reply:
I would just make sure that the pid is also assigned uniquely. A second mapping pid <-> Particle can be used similar to the str one.

Reply:
Ok is checked now 3951fd64. No test the logging though, wonder if it should raise a `UserWarning` or so